core/src/autogluon/core/metrics/__init__.py

```diff
@@ -454,6 +541,11 @@ def get_metric(metric, problem_type=None, metric_type=None) -> Scorer:
             # Requires mxnet
             from .softclass_metrics import soft_log_loss
             return soft_log_loss
+        elif metric == 'pinball_loss':
+            if problem_type is not None and problem_type is not QUANTILE:
+                raise ValueError(f"{metric_type}={metric} can only be used for quantile problems")
+            from .quantile_metrics import pinball_loss
+            return pinball_loss
         else:
```

Comment:
May also want to check that incompatible metric (eg. accuracy) is not being specified when problem_type == QUANTILE

Reply:
Can we also have a check: 
```
if problem_type == QUANTILE and metric not in QUANTILE_METRICS + [None]:
    raise ValueError(f"Only the following metrics can be used for quantile problems: {QUANTILE_METRICS}")
```

Reply:
I just make it clear as in other cases.
Does it look good? 
Here, but only the `soft_log_loss` will be problematic.
```
if metric is not None and isinstance(metric, str):
    if metric in QUANTILE_METRICS:
        if problem_type is not None and problem_type != QUANTILE:
            raise ValueError(f"{metric_type}={metric} can only be used for quantile problems")
        return QUANTILE_METRICS[metric]
    elif metric in CLASSIFICATION_METRICS:
        if problem_type is not None and problem_type not in PROBLEM_TYPES_CLASSIFICATION:
            raise ValueError(f"{metric_type}={metric} can only be used for classification problems")
        return CLASSIFICATION_METRICS[metric]
    elif metric in REGRESSION_METRICS:
        if problem_type is not None and problem_type not in PROBLEM_TYPES_REGRESSION:
            raise ValueError(f"{metric_type}={metric} can only be used for regression problems")
        return REGRESSION_METRICS[metric]
    elif metric == 'soft_log_loss':
        # Requires mxnet
        from .softclass_metrics import soft_log_loss
        return soft_log_loss
    else:
        raise ValueError(
            f"{metric} is an unknown metric, all available metrics are "
            f"'{all_available_metric_names}'. You can also refer to "
            f"autogluon.core.metrics to see how to define your own {metric_type} function"
        )
else:
    return metric

```